electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -57,8 +99,39 @@ def append(
         if event:
             self.events.append(event)
 
+    @staticmethod
+    def merge_exchanges(
+        ungrouped_exchanges: List["ExchangeList"], logger: Logger
+    ) -> "ExchangeList":
+        """
+        Given multiple parser outputs, sum the netflows of corresponding datetimes
+        to create a unique exchange list. Sources will be aggregated in a
+        comma-separated string. Ex: "entsoe, eia".
+        """
+        exchanges = ExchangeList(logger)
+        if ExchangeList.is_completly_empty(ungrouped_exchanges, logger):
+            return exchanges
+
+        # Create a dataframe for each parser output, then flatten the exchanges.
+        exchange_dfs = [
+            pd.json_normalize(exchanges.to_list()).set_index("datetime")
+            for exchanges in ungrouped_exchanges
+            if len(exchanges.events) > 0
+        ]
+
+        exchange_df = pd.concat(exchange_dfs)
+        exchange_df = exchange_df.rename(columns={"sortedZoneKeys": "zoneKey"})
+        zone_key, sources, source_type = ExchangeList.get_unique_zone_source(
+            exchange_df
+        )
+        exchange_df = exchange_df.groupby(level=0, dropna=False).sum()
+        for datetime, row in exchange_df.iterrows():
+            exchanges.append(zone_key, datetime, sources, row["netFlow"], source_type)
```

Comment:
Looks like datetime is not getting the correct type here, I think we need to cast the type first. Right now it's a lable or hashable when it should be a datetime.

Reply:
I think we need to ignore it anyways as we did in the merge production method.

Reply:
I don't think the cast is going to properly work.